conans/test/functional/remote/multi_remote_test.py

```diff
@@ -2,13 +2,96 @@
+        client_b = TestClient(servers=self.servers, users={"default": [("lasote", "mypass")],
+                                                           "local": [("lasote", "mypass")]})
+
+        # Upload Hello0 to local and default from client_a
+        self._create(client_a, "Hello0", "0.0")
+        client_a.run("upload Hello0/0.0@lasote/stable -r local")
+        client_a.run("upload Hello0/0.0@lasote/stable -r default")
+        client_a.run("remote list_ref")
+        self.assertIn(": local", str(client_a.user_io.out))
+        sleep(1)  # For timestamp and updates checks
+
+        # Download Hello0 from local with client_b
+        client_b.run("install Hello0/0.0@lasote/stable -r local --build missing")
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+
+        # Update Hello0 with client_a and reupload
+        self._create(client_a, "Hello0", "0.0", modifier="\n")
+        client_a.run("upload Hello0/0.0@lasote/stable -r local")
+
+        # Execute info method in client_b, should advise that there is an update
+        client_b.run("info Hello0/0.0@lasote/stable -u")
+        self.assertIn("Recipe: Update available", client_b.out)
+        self.assertIn("Binary: Cache", client_b.out)
+
+        # Now try to update the package with install -u
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+        client_b.run("install Hello0/0.0@lasote/stable -u --build")
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+
+        # Upload a new version from client A, but only to the default server (not the ref-listed)
+        # Upload Hello0 to local and default from client_a
+        sleep(1)  # For timestamp and updates checks
+        self._create(client_a, "Hello0", "0.0", modifier="\n\n")
+        client_a.run("upload Hello0/0.0@lasote/stable -r default")
+
+        # Now client_b checks for updates without -r parameter
+        client_b.run("info Hello0/0.0@lasote/stable -u")
+        self.assertIn("    Remote: local", str(client_b.user_io.out))
+        self.assertIn("    Recipe: Cache", client_b.out)
+
+        # But if we connect to default, should tell us that there is an update IN DEFAULT!
+        client_b.run("info Hello0/0.0@lasote/stable -r default -u")
+        self.assertIn("Remote: local", str(client_b.user_io.out))
+        client_b.run("remote list_ref")
+        self.assertIn(": local", str(client_b.user_io.out))
+
+        # Well, now try to update the package with -r default -u
+        client_b.run("install Hello0/0.0@lasote/stable -r default -u --build")
+        self.assertIn("Hello0/0.0@lasote/stable: Calling build()",
+                      str(client_b.user_io.out))
+        client_b.run("info Hello0/0.0@lasote/stable -u")
+        self.assertIn("Recipe: Cache", client_b.out)
+        self.assertIn("Binary: Cache", client_b.out)
```

Comment:
And check the `remote list_ref`... has it changed? is it associated with `local` or `default`?